.github/workflows/_meta-build.yaml

```diff
@@ -78,29 +78,36 @@ jobs:
 
       - name: Login to Docker.io
         uses: docker/login-action@v2.0.0
-        if: ${{ github.ref == 'refs/heads/master' }}
+        if: ${{ github.event_name == 'push' }}
         with:
           registry: docker.io
           username: ${{ secrets.registry-0-usr }}
           password: ${{ secrets.registry-0-psw }}
 
+      - name: Set Container Tags
+        id: tags
+        run: |-
+          TAGS="docker.io/dependencytrack/${{ matrix.distribution }}:${{ inputs.app-version }}"
+          if [[ "${{ inputs.app-version }}" != "snapshot" ]]; then
+            TAGS="${TAGS},docker.io/dependencytrack/${{ matrix.distribution }}:latest"
+          fi
+          echo "::set-output name=tags::${TAGS}"
+
       - name: Build multi-arch Container Image
         uses: docker/build-push-action@v3.0.0
         with:
-          tags: |-
-            docker.io/dependencytrack/${{ matrix.distribution }}:${{ inputs.app-version }}
-            docker.io/dependencytrack/${{ matrix.distribution }}:latest
+          tags: ${{ steps.tags.outputs.tags }}
           build-args: |-
             APP_VERSION=${{ inputs.app-version }}
             COMMIT_SHA=${{ github.sha }}
             WAR_FILENAME=dependency-track-${{ matrix.distribution }}.jar
           platforms: linux/amd64,linux/arm64
-          push: ${{ github.ref == 'refs/heads/master' }}
+          push: ${{ github.event_name == 'push' }}
```

Comment:
Sorry for all the nitpicking, but now I think we may push to `snapshot` from `4.5.x` *and* `master`. 

What we need is something along the lines of:
* `master`: Push releases and snapshots
* `x.x.x`: Push releases only

I feel like it'd make sense to adopt something similar like we discussed in https://github.com/DependencyTrack/frontend/pull/141, which is creating releases based on tags. Thoughts?

Reply:
you are not nitpicking, you are reviewing, and I am thankful for that!

you are in fact correct that would happen + another flaw would be if my assumption that the event name gets passed to the meta workflow would not hold true.